packages/services/pkg/mode/data.go

```diff
@@ -66,6 +84,16 @@ func SerializeRow(row []interface{}, colNames []string, colEncodingTypes []*abi.
 				if err != nil {
 					return nil, err
 				}
+			} else if colEncodingTypes[i].String() == "bytes" || colEncodingTypes[i].String() == "string" {
+				// Handle bytes / string specially to avoid the offset missing issue.
+				encodedField, err = EncodeParameters(
+					[]string{colEncodingType.String()},
+					[]interface{}{row[i]},
+				)
```

Comment:
are there tests for these sorts of things to demonstrate what the expected output is for given input?

Reply:
not yet, good point though. i'll add a bunch when we implement a proper encoding solution to make sure sync works (will be separate pr)